test/libsolidity/Metadata.cpp

```diff
@@ -37,13 +37,18 @@ namespace solidity::frontend::test
 namespace
 {
 
-map<string, string> requireParsedCBORMetadata(bytes const& _bytecode)
+map<string, string> requireParsedCBORMetadata(bytes const& _bytecode, CompilerStack::MetadataFormat _metadataFormat)
 {
 	bytes cborMetadata = solidity::test::onlyMetadata(_bytecode);
-	BOOST_REQUIRE(!cborMetadata.empty());
-	std::optional<map<string, string>> tmp = solidity::test::parseCBORMetadata(cborMetadata);
-	BOOST_REQUIRE(tmp);
-	return *tmp;
+	if (_metadataFormat != CompilerStack::MetadataFormat::NoMetadata)
```

Comment:
Check that it is missing otherwise?

Reply:
There is a check after at line 50:
```
BOOST_REQUIRE(cborMetadata.empty());
return {};
```
if that was what you mean with check?